spec/services/unwithdraw_service_spec.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+RSpec.describe UnwithdrawService do
+  before do
+    @user = create(:user)
+  end
+
+  describe "#call" do
+    it "adds an entry in the timeline of the document" do
+      withdrawn_edition = create(:edition, :withdrawn)
+      stub_publishing_api_republish(withdrawn_edition.content_id, {})
+
+      UnwithdrawService.new.call(withdrawn_edition, @user)
+
+      expect(withdrawn_edition.timeline_entries.first.entry_type).to eq("unwithdrawn")
+    end
+
+    it "creates a new status with the same state as the previously published version" do
+      withdrawn_edition = create(:edition, :withdrawn)
+      stub_publishing_api_republish(withdrawn_edition.content_id, {})
+
+      previously_published_state = withdrawn_edition.status.details.published_status.state
```

Comment:
Perhaps it would also be good to check the statuses are actually different rows here (vs. the behaviour for editing the withdrawing).

Reply:
Yep sure!